components/header/AnimatedHeader.tsx

```diff
@@ -0,0 +1,173 @@
+import React from "react";
+import { View, Text, Image } from "react-native";
+import Animated, {
+  useAnimatedStyle,
+  interpolate,
+  Extrapolation,
+} from "react-native-reanimated";
+import {
+  UnistylesRuntime,
+  createStyleSheet,
+  useStyles,
+} from "react-native-unistyles";
+import { useWeatherData } from "../../hooks/useWeatherData";
+import { SharedValue } from "react-native-reanimated";
+
+const AnimatedHeader: React.FC<{
+  city: string;
+  today: boolean;
+  scrollY: SharedValue<number>;
+  headerHeight: number;
+}> = ({ city, today, scrollY, headerHeight }) => {
+  const { styles } = useStyles(stylesheet);
+  const {
+    minTemp,
+    maxTemp,
+    mainWeather,
+    mainTemp,
+    forecastData,
+    getDayWeatherSummary,
+  } = useWeatherData(city, today);
+  const insets = UnistylesRuntime.insets;
+
+  const animatedStyle = useAnimatedStyle(() => {
+    const translateY = interpolate(
+      scrollY.value,
+      [0, headerHeight],
+      [-headerHeight, -insets.top],
+      Extrapolation.CLAMP,
+    );
+
+    return {
+      transform: [{ translateY }],
+    };
+  });
+
+  const mainInfoBoxAnimatedStyle = useAnimatedStyle(() => {
+    const opacity = interpolate(
+      scrollY.value,
+      [0, headerHeight / 2],
+      [1, 0],
+      Extrapolation.CLAMP,
+    );
+    return { opacity };
+  });
+
+  const { mainIcon } = getDayWeatherSummary(0);
+  const weatherIcon =
+    today && forecastData ? forecastData.list[0].weather[0].icon : mainIcon;
+
+  return (
+    <>
+      <Animated.View style={[styles.mainInfoBox, mainInfoBoxAnimatedStyle]}>
+        <Text style={styles.city}>{city}</Text>
+        <Text style={styles.mainTemp}>{mainTemp}</Text>
+        <Text style={styles.weatherDescription}>{mainWeather}</Text>
+        <Text style={styles.minMax}>
+          from {minTemp?.toFixed() + "°"} to {maxTemp?.toFixed() + "°"}
+        </Text>
+      </Animated.View>
+
+      <Animated.View style={[styles.animationBox, animatedStyle]}>
+        <View style={styles.animatedMainInfoBox}>
+          <View style={styles.animatedWeatherInfo}>
+            <View style={styles.animatedCityBox}>
+              <Text style={styles.animatedCity}>{city}</Text>
+            </View>
+            <View style={styles.animatedMainWeatherBox}>
+              <Text style={styles.animatedMainTemp}>{mainTemp}</Text>
+              <Image
+                source={{
+                  uri: `https://openweathermap.org/img/wn/${weatherIcon}@2x.png`,
+                }}
+                style={[styles.weatherIcon, styles.shadow]}
+              />
+            </View>
+          </View>
+        </View>
+      </Animated.View>
```

Comment:
Czemu ten komponent ma w sobie 2 zupełnie niezależne od siebie widoki? To ewidentnie powinny być 2 komponenty. Nie musiałbyś też robić jakichś workaroundów z tym `stickyHeaderIndices` w miejscu, gdzie tego używasz

Reply:
To rowniez rozwiazane w nowszym branchu